src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -54,4 +59,44 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     final protected String parserName() {
         return MatchAllQueryParser.NAME;
     }
-}
\ No newline at end of file
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        if (this.boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
+        query.setBoost(boost);
+        return query;
+    }
+
+    @Override
+    public int hashCode() {
+        return Float.hashCode(this.boost);
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        this.boost = in.readFloat();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeFloat(this.boost);
+    }
+
+    @Override
```

Comment:
Sure, great idea. Btw. I looked at some of the java.utils/guava helpers for equals, etc. because I'd like to get rid of the boilerplate in the first two if-statements here, but could not find anything. Any ideas?